src/server/chunk_server/chunk_server_impl.h

```diff
@@ -0,0 +1,95 @@
+#ifndef GFS_SERVER_CHUNK_SERVER_IMPL_H_
+#define GFS_SERVER_CHUNK_SERVER_IMPL_H_
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/time/time.h"
+#include "google/protobuf/stubs/statusor.h"
+#include "grpcpp/grpcpp.h"
+#include "src/common/config_manager.h"
+#include "src/common/protocol_client/chunk_server_service_server_client.h"
+#include "src/common/protocol_client/master_chunk_server_manager_service_client.h"
+#include "src/common/utils.h"
+
+namespace gfs {
+namespace server {
+
+class ChunkServerImpl {
+ public:
+  static google::protobuf::util::StatusOr<ChunkServerImpl*>
+  ConstructChunkServerImpl(const std::string& config_filename,
+                           const std::string& chunk_server_name,
+                           const bool resolve_hostname = false);
+
+  // Add or update the write lease for |file_handle| to expire at UNIX seconds
+  // timestamp of |expiration_unix_sec|.
+  void AddOrUpdateLease(const std::string& file_handle,
+                        const uint64_t expiration_unix_sec);
+  void RemoveLease(const std::string& file_handle);
+  bool HasWriteLease(const std::string& file_handle);
+  // Return NOT_FOUND, if no lease exists for file handle
+  google::protobuf::util::StatusOr<absl::Time> GetLeaseExpirationTime(
+      const std::string& file_handle);
+
+  // MOCK FUNCTIONS for chunk file manager
```

Comment:
nit: more functions for CRUD. I believe that will come in next?

Reply:
I plan to use your file chunk manager now. so these can go away and no more mocks necessary